war/src/main/js/tooltips.js

```diff
@@ -0,0 +1,91 @@
+import tippy from "tippy.js"
+
+const TOOLTIP_BASE = {
+  arrow: false,
+  theme: "tooltip",
+  animation: "tooltip",
+  appendTo: document.body
+}
+
+let tooltipInstances = []
+const globalPlugin = {
+  fn(instance) {
+    return {
+      onCreate() {
+        tooltipInstances = tooltipInstances.concat(instance)
+      },
+      onDestroy() {
+        tooltipInstances = tooltipInstances.filter(i => i !== instance)
+      }
+    }
+  }
+}
+
+tippy.setDefaultProps({
+  plugins: [globalPlugin]
+})
+
+registerTooltips()
+
+/**
+ * Registers tooltips for the page
+ * If called again, destroys existing tooltips and registers them again (useful for progressive rendering)
+ * @param {HTMLElement} container - Registers the tooltips for the given container
+ */
+function registerTooltips(container) {
+  if (!container) {
+    container = document
+  }
+
+  tooltipInstances.forEach(instance => {
+    if (instance.props.container === container) {
+      instance.destroy()
+    }
+  })
+
+  tippy(container.querySelectorAll("[tooltip]:not([tooltip=\"\"])"), Object.assign({
```

Comment:
Could this be made conditional on no `html-tooltip` being set? This came up in matrix-auth, and while Tim provided a workaround, it would seem to make the transition easier if I can set both attributes.

* Current Jenkins uses `tooltip` to render it as HTML, ignoring `html-tooltip`.
* Future Jenkins uses `html-tooltip` to render it as HTML, ignoring (badly formatted) `tooltip`.

Reply:
done